app/models/volunteer.rb

```diff
@@ -124,66 +129,77 @@ def has_main_region?
   end
 
   def main_region
-    self.regions[0]
+    regions[0]
   end
 
   def region_ids
-    self.regions.collect{ |r| r.id }
+    regions.collect(&:id)
   end
 
-  def admin_region_ids(strict=false)
-    admin_regions(strict).collect { |r| r.id }
+  def admin_region_ids(strict = false)
+    admin_regions(strict).collect(&:id)
   end
 
-  def admin_regions(strict=false)
-    if self.super_admin? and not strict
+  def admin_regions(strict = false)
+    if super_admin? && !strict
       Region.all
     else
-      self.assignments.collect{ |a| a.admin ? a.region : nil }.compact
+      assignments.collect { |a| a.admin ? a.region : nil }.compact
```

Comment:
Can
`assignments.collect { |a| a.admin ? a.region : nil }.compact`
be
`assignments.select { |a| a.admin && a.region }`